splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/actions/CreateIndexConstantOperation.java

```diff
@@ -452,6 +452,9 @@ public void executeConstantAction( Activation activation ) throws StandardExcept
                 }
 
                 IndexRowGenerator irg = cd.getIndexDescriptor();
+                if (irg.isUnique() != unique) {
```

Comment:
Could you also add check so that if  excludeNulls or excludeDefaults do not match, the index cannot be shared either?
Here is a test case:
```
create table b(i int, j int, primary key (i));
create table a (i int, j int);
insert into b values (1,1);
insert into a values (1,1);
create index aj on a(j) exclude null keys;
alter table a add CONSTRAINT fc FOREIGN KEY(j) REFERENCES b(i) ON UPDATE NO ACTION ON DELETE restrict;
splice> show indexes from a;
TABLE_NAME                                        |INDEX_NAME                                        |COLUMN_NAME         |ORDINAL&|NON_UNIQUE|TYPE |ASC&|CONGLOM_NO
--------------------------------------------------------------------------------------------------------------------------------------------------------------------
A                                                 |AJ                                                |J                   |1       |true      |BTREE|A   |1665      
A                                                 |SQL200701155616910                                |J                   |1       |true      |BTREE|A   |1665      

2 rows selected
ELAPSED TIME = 1531 milliseconds
```